app/frontend/src/types.ts

```diff
@@ -1,50 +1,64 @@
+// Represents a grounding file
 export type GroundingFile = {
     id: string;
     name: string;
     content: string;
 };
 
+// Represents an item in the history
 export type HistoryItem = {
     id: string;
     transcript: string;
     groundingFiles: GroundingFile[];
+    sender: "user" | "assistant"; // Add sender field
+    timestamp: Date; // Add timestamp field
 };
 
+// Represents a command to update the session
 export type SessionUpdateCommand = {
     type: "session.update";
     session: {
         turn_detection?: {
             type: "server_vad" | "none";
+            threshold?: number;
+            prefix_padding_ms?: number;
+            silence_duration_ms?: number;
         };
         input_audio_transcription?: {
             model: "whisper-1";
         };
     };
 };
 
+// Represents a command to append audio to the input buffer
 export type InputAudioBufferAppendCommand = {
     type: "input_audio_buffer.append";
-    audio: string;
+    audio: string; // Ensure this is a valid base64-encoded string
 };
 
+// Represents a command to clear the input audio buffer
 export type InputAudioBufferClearCommand = {
     type: "input_audio_buffer.clear";
 };
 
+// Represents a generic message
 export type Message = {
     type: string;
 };
 
+// Represents a response containing an audio delta
 export type ResponseAudioDelta = {
     type: "response.audio.delta";
-    delta: string;
+    delta: string; // Ensure this is a valid base64-encoded string
```

Comment:
This is a response, we can't really ensure what it has vs InputAudioBufferAppendCommand